readyplayerme/rpm-unreal-next-gen-sdk
# Release/v1.0.0

## [TICKETID](https://ready-player-me.atlassian.net/browse/TICKETID)

## Description

This release includes:
- Big refactor of WebApi base class (and subclasses)
- Updated and refactored developer window
- Support for removing assets 
- All skeletal mesh saving functionality now included in SDK directly (previously it was using the 3rd party plugin TransientObjectSaver)

I made this a 1.0 release it includes breaking changes from the previous but also I think it should be much more stable then previous ones. 




## Changes

### Updated

- Developer window now displays error messages and input field validation errors [#19](https://github.com/readyplayerme/rpm-unreal-next-gen-sdk/pull/19)
- WebApi base class and subclasses reworked to improve code quality, request cancellation and error handling [#18](https://github.com/readyplayerme/rpm-unreal-next-gen-sdk/pull/18)

### Added

- Remove asset support added to RPMActor and samples [#17](https://github.com/readyplayerme/rpm-unreal-next-gen-sdk/pull/17)
- SkeletalMesh import functionality added [#20](https://github.com/readyplayerme/rpm-unreal-next-gen-sdk/pull/20)



## How to Test

Developer login window Functionality
- login with username and password (also check invalid it should now provide some feedback and validation) 
- Login with Demo
- Fetching of character styles and importing them (if clicked on button) 
- Logout should work

Testing new WebAPI updates
- simplest way is to run the 2 sample scenes (provided you have already logged in from Developer window (you can also test what happens if you aren't logged in, it won't work but it shouldn't crash the editor) 
- equipping and removing assets (remove by clicking the same asset again) 
- changing character styles with clothing assets equipped (it should reload with correctly refitted assets on new style) 
- Test sample in build if possible 





## Checklist


## QA Testing

-   Mention any useful test cases for this feature and add to QA Testing documentation.

## Details

-   If more details are necessary to support the description with images and videos add them here.
